### PR TITLE
NSG resource Fixes and Webapp resources added

### PR DESCRIPTION
**Pull request by skelleyton**
### Description

Fixes in azurerm_network_security_group to property scan rules that have multiple source IP addresses and multiple destination ports.

Resources added for Webapp and Webapps. Added POST HTTP method to use the authentication settings call for Webapps. 

### Issues Resolved

https://github.com/inspec/inspec-azure/issues/169

### Check List

- [ x] New functionality includes tests
- [ x] All tests pass
- [ x] `rake lint` passes
- [ x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by davymcaleer** [comment]
@skelleyton aiming to get one of the team onto this PR in the next 2 weeks

**Comment by mikaelkrief** [comment]
Thanks - I am also very interested with this PR for use inspec for web app

**Comment by balasundarbala** [comment]
It can handle function app as well ?

**Comment by skelleyton** [comment]
This PR does not handle Functions, just Web Apps.

File: libraries/azurerm_webapps.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'azurerm_resource'
+
+class AzurermWebapps < AzurermPluralResource
+  name 'azurerm_webapps'
+  desc 'Verifies settings for Webapps'
+  example <<-EXAMPLE
+    azurerm_webaps(resource_group: 'example') do
```
Comment (r-fennell) [inline libraries/azurerm_webapps.rb]: Minor typo

File: docs/resources/azurerm_webapp.md.erb [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+An `azurerm_webapp` resource block identifies an AppService Webapp by name and Resource Group.
+
+    describe azurerm_appservice_webapp(resource_group: 'example', name: 'webapp name') do
+      ...
+    end
+
+<br />
+
+## Examples
```
Comment (r-fennell) [inline docs/resources/azurerm_webapp.md.erb]: Would be great to see examples of the additional resource methods added here, for example
```
its('auth_settings.properties') { should have_attributes(enabled: false) }
its('configuration.properties') { should have attributes(netFrameworkVersion: 'v4.0') }
```

File: test/integration/verify/controls/azurerm_webapps.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+resource_group = attribute('resource_group', default: nil)
```
Comment (r-fennell) [inline test/integration/verify/controls/azurerm_webapps.rb]: You've already exposed `webapp_name` for the singular resource, so best to make use of it here too.
`webapp_name = attribute('webapp_name', default: nil)`
then
`     its('names')                { should include webapp_name } `

**Comment by r-fennell** [comment]
Thanks @skelleyton

**Pull request merged**